front/src/applications/stdcm/utils/__tests__/simulationReportSheet.spec.ts

```diff
@@ -0,0 +1,80 @@
+import {
+  generateCodeNumber,
+  formatCreationDate,
+  getStopDurationTime,
+  extractSpeedLimit,
+  getOperationalPointsWithTimes,
+} from 'applications/stdcm/utils/formatSimulationReportSheet';
+
+import { suggestedOperationalPoints, simulation, departure_time } from './sampleData';
+
+describe('Utils simulation report sheet functions', () => {
+  test('should return a formatted string', () => {
+    const codeNumber = generateCodeNumber();
+    expect(codeNumber).toMatch(/^\d{2}\d{2}-\d{3}-\d{3}$/);
+  });
+
+  test('should format the date correctly', () => {
+    const formattedDate = formatCreationDate(departure_time);
+    expect(formattedDate).toEqual({
+      day: '08',
+      month: '07',
+      year: 2024,
+      hours: 10,
+      minutes: 20,
+    });
+  });
+
+  test('should return correct time format', () => {
+    expect(getStopDurationTime(30)).toBe('30 sec');
+    expect(getStopDurationTime(120)).toBe('2 min');
+  });
+
+  test('should extract the speed limit correctly', () => {
+    const speedLimitByTag = 'Some tag - 50';
+    expect(extractSpeedLimit(speedLimitByTag)).toBe('50');
```

Comment:
Could you add a test for the tags with syntax "Voyageurs" or "Marchandises" ? You can use any name but it should be without a " - " to be sure it's working

Reply:
Done